staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,379 @@
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status
+	// +optional
+	Status PriorityLevelConfigurationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	// `metadata` is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// `items` is a list of request-priorities.
+	Items []PriorityLevelConfiguration `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// `queuingType` indicates whether this priority level does
+	// queuing or is exempt.  Valid values are "Queuing" and "Exempt",
+	// and this field defaults to "Queuing".  "Exempt" means that
+	// requests of this priority level are not subject to concurrency
+	// limits (and thus are never queued) and do not detract from the
+	// concurrency available for non-exempt requests.
+	// +optional
+	QueuingType PriorityLevelQueueingType `json:"queuingType,omitempty" protobuf:"varint,1,opt,name=queuingType"`
+
+	// `queuingConfig` holds the configuration parameters that are
+	// only meaningful for a priority level that does queuing (i.e.,
+	// is not exempt).  This field must be non-empty if and only if
+	// `queuingType` is `"Queuing"`.
+	// +optional
+	QueuingConfig *QueuingConfiguration `json:"queuingConfig,omitempty" protobuf:"bytes,2,opt,name=queuingConfig"`
+}
+
+// PriorityLevelQueueingType identifies the queuing nature of a priority level
+type PriorityLevelQueueingType string
+
+// PriorityLevelQueuingTypeQueueing is the PriorityLevelQueueingType for priority levels that queue
+const PriorityLevelQueuingTypeQueueing PriorityLevelQueueingType = "Queuing"
+
+// PriorityLevelQueuingTypeExempt is the PriorityLevelQueueingType for priority levels that are exempt from concurrency controls
+const PriorityLevelQueuingTypeExempt PriorityLevelQueueingType = "Exempt"
+
+// QueuingConfiguration holds the configuration parameters that are specific to a priority level that is subject to concurrency controls
+type QueuingConfiguration struct {
+	// `assuredConcurrencyShares` (ACS) must be a positive number. The
+	// API server's concurrency limit (SCL) is divided among the
+	// concurrency-controlled priority levels in proportion to their
+	// assured concurrency shares. This produces the assured
+	// concurrency value (ACV) for each such priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	// Required.
```

Comment:
This is the last required field here.  By scaling a default here to be a reasonable percentage of our "bootstrap policies", I think we can make this optional with a default set.

Reply:
I do not understand what you are suggesting.  Can you please elaborate?

Reply:
@deads2k: on re-reading I think maybe I understand what you mean.  The "last" in your first sentence refers to progression in development time.  You are suggesting that we define a default value for this field, by choosing a value that bears a non-sucky relation with the ACS values in the cluster initial configuration.  I think that makes sense.  We have two versions of the cluster initial configuration: one in the KEP (https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md#example-configuration) and one in the feature branch (https://github.com/kubernetes/kubernetes/blob/feature-rate-limiting/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/bootstrap/default-objects.go#L95).  For both of them, I think a good default value would be `30`.

Reply:
I think David is suggesting a default, and I agree that this is useful, if for no other reason than to anchor people on the right order-of-magnitude. 100? 1000? It depends on the rest of the objects in the system, which is the default objects at first.

Reply:
It sounds like @lavalamp is agreeing with my interpretation of @deads2k's comment.  I think a default of `30` _does_ anchor readers in the right order-of-magnitude, I chose it after reviewing the cluster initial configuration ("default objects", if you will --- but I won't, because we have used the words "default objects" to mean various things, see the KEP updates about that).

Reply:
30 sounds fine for our first take.

Reply:
(I hadn't refreshed my page and hadn't seen your 2nd comment when I wrote my comment.)

Reply:
applied 30 as its default value, now there's no required value in the priority-level's spec

Reply:
The comments that I see now still say this required rather than defaulted.  Did you push your latest changes?

Reply:
Yeah it is showing as required with no mention of the default to me, too?